server.go

```diff
@@ -127,13 +127,13 @@ func (s *Server) Serve(ctx context.Context, l net.Listener) error {
 
 func (s *Server) Shutdown(ctx context.Context) error {
 	s.mu.Lock()
-	lnerr := s.closeListeners()
 	select {
 	case <-s.done:
 	default:
 		// protected by mutex
 		close(s.done)
 	}
+	lnerr := s.closeListeners()
```

Comment:
Hm, what is the difference?

Reply:
not closing until the server is done. The CI still has some weird errors that i cannot repo locally, that is why this has been sitting